src/GLX/libglxmapping.c

```diff
@@ -539,6 +520,16 @@ __GLXvendorInfo *__glXLookupVendorByName(const char *vendorName)
                 goto fail;
             }
 
+            // The vendor must provide both or neither of
+            // getContextDispatchHandle and getContextProcAddress.
+            if (pEntry->imports.getContextDispatchHandle != NULL
+                    || pEntry->imports.getContextProcAddress != NULL) {
+                if (pEntry->imports.getContextDispatchHandle == NULL
+                        || pEntry->imports.getContextProcAddress == NULL) {
```

Comment:
I expect some people might not like the number of `!`, but I'd find that easier to read:
```
if (!pEntry->imports.getContextDispatchHandle !=
    !pEntry->imports.getContextProcAddress)
```